src/main/java/org/isf/usersettings/rest/UserSettingController.java

```diff
@@ -0,0 +1,203 @@
+	 * @throws OHServiceException
+	 */
+	@PutMapping(value = "/usersettings/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
+	public UserSettingDTO updateUserSettings(@PathVariable(name = "id") int id, @Valid @RequestBody UserSettingDTO userSettingDTO)
+					throws OHServiceException {
+		LOGGER.info("Update a UserSetting.");
+		String requestUserName = userSettingDTO.getUser();
+		String currentUser = SecurityContextHolder.getContext().getAuthentication().getName();
+		if (userSettingDTO.getId() == 0 || userSettingDTO.getId() != 0 && userSettingDTO.getId() != id) {
+			throw new OHAPIException(new OHExceptionMessage("Malformed request."));
+		}
+		Optional<UserSetting> userSetting = userSettingManager.getUserSettingById(id);
+		UserSetting updated;
+		if (userSetting.isEmpty()) {
+			LOGGER.info("No user settings with id {}.", id);
+			throw new OHAPIException(new OHExceptionMessage("UserSetting doesn't exists."));
+		}
+		if (!userSetting.get().getUser().equals(requestUserName) || !userSetting.get().getUser().equals(currentUser)) {
+			throw new OHAPIException(new OHExceptionMessage("Not allowed."), HttpStatus.FORBIDDEN);
+		}
+		if (userManager.getUserByName(requestUserName) == null) {
+			throw new OHAPIException(new OHExceptionMessage("The specified user does not exist."));
+		}
+		UserSetting uSetting = userSettingMapper.map2Model(userSettingDTO);
+		updated = userSettingManager.updateUserSetting(uSetting);
+		if (updated == null) {
+			LOGGER.info("UserSetting is not updated.");
+			throw new OHAPIException(new OHExceptionMessage("UserSetting not updated."));
+		}
+		LOGGER.info("UserSetting successfully updated.");
+		return userSettingMapper.map2DTO(updated);
+	}
+
+	/**
+	 * Retrieves an existing {@link UserSettingDTO} by user.
+	 * @param configName - the name of the userSetting {@link UserSetting} .
+	 * @return {@link UserSettingDTO} if the UserSetting exists, null otherwise.
+	 * @throws OHServiceException
+	 */
+	@GetMapping(value = "/usersettings/{configName}", produces = MediaType.APPLICATION_JSON_VALUE)
+	public UserSettingDTO getUserSettingByUser(@PathVariable(name = "configName") String configName) throws OHServiceException {
+		String currentUser = SecurityContextHolder.getContext().getAuthentication().getName();
+		LOGGER.info("Retrieve the userSetting By user {} and configName {}.", currentUser, configName);
+		UserSetting userSetting = userSettingManager.getUserSettingByUserNameConfigName(currentUser, configName);
+		if (userSetting == null) {
+			LOGGER.info("No user settings '{}' for the user {}.", configName, currentUser);
+			throw new OHAPIException(new OHExceptionMessage("No setting found"), HttpStatus.NOT_FOUND);
+		}
+		return userSettingMapper.map2DTO(userSetting);
+	}
+
+	/**
+	 * Deletes a {@link UserSetting}.
+	 * @param id - the id of the userSetting {@link UserSetting} to delete.
+	 * @return {@code true} if the userSetting has been deleted, {@code false} otherwise.
```

Comment:
where is `false` returned in this method?  All I see is one `true` and the rest are exceptions.